File: ios/Counter.m
//
//  Counter.m
//  gemlestest
//
//  Created by Colin Dively on 4/9/23.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"


@interface RCT_EXTERN_MODULE(Counter,NSObject)
RCT_EXTERN_METHOD(increment:(RCTResponseSenderBlock)callback)
@end

@interface RCT_EXTERN_MODULE(GemViewManager, RCTViewManager)
@end

File: ios/Counter.swift
//
//  Counter.swift
//  gemlestest
//
//  Created by Colin Dively on 4/9/23.
//

import Foundation

@objc(Counter)
class Counter: NSObject {
  private var count = 0
  
  @objc
  func increment(_ callback:RCTResponseSenderBlock) {
     count += 1
//    print(count)
    callback([count])
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func constantsToExport() -> [String: Any]! {
    return ["initialCount": 0]
  }
  
}
